source/extensions/common/aws/utility.cc

```diff
@@ -223,19 +223,18 @@ static size_t curlCallback(char* ptr, size_t, size_t nmemb, void* data) {
   return nmemb;
 }
 
-absl::optional<std::string> Utility::metadataFetcher(const std::string& host,
-                                                     const std::string& path,
-                                                     const std::string& auth_token) {
+absl::optional<std::string> Utility::fetchMetadata(Http::RequestMessage& message) {
   static const size_t MAX_RETRIES = 4;
   static const std::chrono::milliseconds RETRY_DELAY{1000};
   static const std::chrono::seconds TIMEOUT{5};
 
   CURL* const curl = curl_easy_init();
-  if (!curl) {
-    return absl::nullopt;
-  };
+  RELEASE_ASSERT(curl != nullptr, "");
+  const auto host = message.headers().getHostValue();
+  const auto path = message.headers().getPathValue();
+  const auto scheme = message.headers().getSchemeValue();
 
-  const std::string url = fmt::format("http://{}/{}", host, path);
+  const std::string url = fmt::format("{}://{}{}", scheme, host, path);
```

Comment:
Would "https" be a thing here?  I get your intent, just wondering if it's overkill :)

Reply:
As of now the scheme will just be `http` until we introduce the WebIdentity Credentials Provider, where it will be `https`.
As you suggest I will remove setting scheme part for now. We can reintroduce setting scheme along with WebIdentity Credentials provider. Which will be after we do the http async client part.